enterprise/app/workflows/workflows.tsx

```diff
@@ -289,60 +281,70 @@ class ListWorkflowsComponent extends React.Component<ListWorkflowsProps, State>
               ))}
               {/* Render legacy workflows */}
               {this.state.workflowsResponse?.workflow.map((workflow) => (
-                <RepoItem
-                  repoUrl={workflow.repoUrl}
-                  webhookUrl={workflow.webhookUrl}
-                  onClickUnlinkItem={() => this.setState({ workflowToDelete: workflow })}
-                />
+                <>
+                  <RepoItem
+                    isAdmin={isAdmin}
+                    repoUrl={workflow.repoUrl}
+                    webhookUrl={workflow.webhookUrl}
+                    onClickUnlinkItem={() => this.setState({ workflowToDelete: workflow })}
+                  />
+                  {workflow.repoUrl && this.renderActionList(workflow.repoUrl)}
+                </>
               ))}
             </div>
           )}
-          <SimpleModalDialog
-            title="Unlink repository"
-            isOpen={Boolean(this.state.repoToUnlink)}
-            onRequestClose={() => this.setState({ repoToUnlink: null })}
-            submitLabel="Unlink"
-            destructive
-            onSubmit={() => this.onClickUnlinkRepo()}
-            loading={this.state.isUnlinkingRepo}>
-            <p>
-              Are you sure you want to unlink <strong>{formatURL(this.state.repoToUnlink || "")}</strong>? This will
-              prevent BuildBuddy workflows from being run.
-            </p>
-          </SimpleModalDialog>
-          <SimpleModalDialog
-            title="Unlink repository"
-            isOpen={Boolean(this.state.workflowToDelete)}
-            onRequestClose={() => this.setState({ workflowToDelete: null })}
-            submitLabel="Unlink"
-            destructive
-            onSubmit={() => this.onClickUnlinkWorkflow()}
-            loading={this.state.isDeletingWorkflow}>
-            <p>
-              Are you sure you want to unlink <strong>{formatURL(this.state.workflowToDelete?.repoUrl || "")}</strong>?
-              This will prevent BuildBuddy workflows from being run.
-            </p>
-          </SimpleModalDialog>
-          <SimpleModalDialog
-            title="Run workflow in clean container"
-            isOpen={Boolean(this.state.showCleanWorkflowWarning)}
-            onRequestClose={() => this.setState({ showCleanWorkflowWarning: false })}
-            submitLabel="Okay"
-            destructive
-            onSubmit={() => this.setState({ showCleanWorkflowWarning: false })}>
-            <p>Are you sure you want to run the workflow in a clean container?</p>
-            <p>
-              This will prevent all existing workflow containers from being reused by other workflow runs, making them
-              slower, so this flag is not encouraged.
-            </p>
-          </SimpleModalDialog>
+          {isAdmin && (
```

Comment:
nit: for simplicity, we can keep the modals rendering even if `!isAdmin`, since the controls to show them are invisible and the worst case is that the modal is somehow visible but the RPC is denied.